src/v/container/tests/fragmented_vector_test.cc

```diff
@@ -8,36 +8,51 @@
  * the Business Source License, use of this software will be governed
  * by the Apache License, Version 2.0
  */
-#define BOOST_TEST_MODULE container
 
 #include "container/fragmented_vector.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
 #include "random/generators.h"
-#include "serde/serde.h"
+#include "serde/rw/rw.h"
+#include "serde/rw/vector.h"
 
-#include <boost/test/tools/old/interface.hpp>
-#include <boost/test/unit_test.hpp>
+#include <gtest/gtest.h>
 
+#include <algorithm>
+#include <cstdint>
 #include <initializer_list>
 #include <limits>
+#include <memory>
 #include <numeric>
+#include <span>
+#include <stdexcept>
 #include <type_traits>
 #include <vector>
 
-using fv_int = fragmented_vector<int>;
+using testing::AssertionFailure;
+using testing::AssertionResult;
+using testing::AssertionSuccess;
+using vec = fragmented_vector<int>;
 
-static_assert(std::forward_iterator<fv_int::iterator>);
-static_assert(std::forward_iterator<fv_int::const_iterator>);
+static_assert(std::forward_iterator<vec::iterator>);
+static_assert(std::forward_iterator<vec::const_iterator>);
 
-namespace test_details {
-
-struct fragmented_vector_accessor {
+class fragmented_vector_validator {
+public:
     // perform an internal consistency check of the vector structure
     template<typename T, size_t S>
-    static void check_consistency(const fragmented_vector<T, S>& v) {
-        BOOST_REQUIRE(v._size <= v._capacity);
-        BOOST_REQUIRE(v.size() < std::numeric_limits<size_t>::max() / 2);
-        BOOST_REQUIRE(v._capacity < std::numeric_limits<size_t>::max() / 2);
-
+    static AssertionResult validate(const fragmented_vector<T, S>& v) {
+        if (v._size > v._next_fragment_needed_size) {
```

Comment:
Not a request for change, but does gtest not have handy assertions which can check > and then output a good error message which also includes the values that failed?

Reply:
Not really in "helper" validation, otherwise if you're in a `TEST` macro you can directly use `EXPECT_LT`, etc, but for building helpers you're on the hook for all the logic, which is a little annoying, but at least AssertionResult has good support for usage in `EXPECT_TRUE(...)` .

See https://chromium.googlesource.com/external/github.com/google/googletest/+/HEAD/docs/advanced.md#predicate-assertions-for-better-error-messages